sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/Uri.java

```diff
@@ -183,7 +184,25 @@ public boolean shouldRefreshHealthStatus() {
     }
 
     public String getHealthStatusDiagnosticString() {
-        return this.healthStatusTuple.get().diagnsoticString;
+        return this.healthStatusTuple.get().getDiagnsoticString();
```

Comment:
This seems to likely be a perf regression because the diagnostics string is generated every single time. Also, a typo in the method name. My hunch is instead having to lazily pre-calculated strings in Uri is better. But ultimately it really depends on how often GetHealthStatusDiagnosticString is called on average during the life cycle. Please check-in with Annie - she will have better info on what realistic expectations are here.,

Reply:
Changed to be precalculated strings.